packages/jest-snapshot/src/inline_snapshots.ts

```diff
@@ -8,8 +8,14 @@
 import * as fs from 'fs';
 import * as path from 'path';
 import semver = require('semver');
+import {parseSync} from '@babel/core';
```

Comment:
`@babel/core` is an optional peer dep, so we need to lazily import it, and gracefully handle it missing ("gracefully" probably just means to throw a more user friendly error saying "install babel or prettier")

Reply:
it missing will probably create some peer dep warnings from the other babel deps... not sure

Reply:
Hmm, thinking about it, why should it be optional? We want the parser either way?

Reply:
Parser yeah, but from my understanding there should only ever exist a single @babel/core. But maybe that's only when loading user config, and we can just depend on core here since we don't actually care about the user's Babel config?

Reply:
We do care about the user's Babel config actually, there might be some early stage syntax in there. But I think it's fine to have a different Babel instance here. Could you elaborate on why for `babel-jest` and other packages it's important to have the same one?

Reply:
Multiple versions of `core` has caused some super weird parsing errors previously, where one tool loaded one version behaving a certain way, and another loaded some other version behaving differently. See #4162. While that was for babel 6&7 specifically, the general idea still applies - the user should decide the version of core, not us.

Might not be needed here though, not sure...

Reply:
Hmm that's a bit of a weird one - it's cool that a user can use Babel 8 and its plugins without this thing trying to use Babel 7 with Babel 8 plugins, but then again there's a chance it just straight up won't work at all, because Babel's core interface itself changed. We don't *really* support future Babel majors yet.
I guess `peerDep` and `dep` (which may have a different minor than the user,  one of them may need to be upgraded for parser/other features/fixes) would both be valid here. But it shouldn't be optional.

Reply:
It's a mandatory peer dep now, I think this should be fine